src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts

```diff
@@ -230,6 +234,27 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 		return undefined;
 	}
 
+	// Fetch any extension environment additions and apply them
+	private async _refreshEnvironmentVariables(activeWorkspaceRootUri: URI | undefined, shellLaunchConfig: IShellLaunchConfig): Promise<void> {
+		const platformKey = platform.isWindows ? 'windows' : (platform.isMacintosh ? 'osx' : 'linux');
+		const lastActiveWorkspace = activeWorkspaceRootUri ? withNullAsUndefined(this._workspaceContextService.getWorkspaceFolder(activeWorkspaceRootUri)) : undefined;
+		const envFromConfigValue = this._workspaceConfigurationService.inspect<ITerminalEnvironment | undefined>(`terminal.integrated.env.${platformKey}`);
+		const isWorkspaceShellAllowed = this._configHelper.checkWorkspaceShellPermissions();
+		this._configHelper.showRecommendations(shellLaunchConfig);
+		const baseEnv = this._configHelper.config.inheritEnv ? processEnv : await this._terminalInstanceService.getMainProcessParentEnv();
+		const env = terminalEnvironment.createTerminalEnvironment(shellLaunchConfig, envFromConfigValue, terminalEnvironment.createVariableResolver(lastActiveWorkspace, this._configurationResolverService), isWorkspaceShellAllowed, this._productService.version, this._configHelper.config.detectLocale, baseEnv);
+
+		if (!shellLaunchConfig.strictEnv) {
+			this._extEnvironmentVariableCollection = this._environmentVariableService.mergedCollection;
+			this._register(this._environmentVariableService.onDidChangeCollections(newCollection => this._onEnvironmentVariableCollectionChange(newCollection)));
+			this._extEnvironmentVariableCollection.applyToProcessEnvironment(env);
```

Comment:
Looks like this is already happening for the remote here:

https://github.com/microsoft/vscode/blob/821afe5e92921ced2461d28c27acd19b02aae68a/src/vs/workbench/api/node/extHostTerminalService.ts#L214-L218

Is it just the event forwarding we're missing?

Reply:
If you look into it, it's more than just forwarding that happens. I didn't think we should copy all of that code into that `extHostTerminalService` and `remoteTerminalChannel`. 
the `_onEnvironmentVariableCollectionChange` method, for instance, with its dependencies.